src/tools/spectest-interp.cc

```diff
@@ -288,17 +289,29 @@ bool ValidIR(const std::string& filename) {
   ReadBinaryOptions options(s_features, s_log_stream.get(), kReadDebugNames,
                             kStopOnFirstError, kFailOnCustomSectionError);
 
-  Errors errors;
   wabt::Module module;
+  Errors errors;
   if (Failed(ReadBinaryIr(filename.c_str(), file_data.data(), file_data.size(),
                           options, &errors, &module))) {
     return false;
   }
 
+  if (!validate) {
+    return true;
+  }
+
   return Succeeded(
       ValidateModule(&module, &errors, ValidateOptions{s_features}));
 }
 
+bool WellformedIR(const std::string& filename) {
+  return CheckIR(filename, false);
```

Comment:
Is "well formed" something that exists in the spec, or something we a creating there?

Reply:
It seems to be the spec's term (e.g. "A byte sequence is a well-formed encoding of a module if and only if it is generated by the grammar." from 5.1). If you think it's better to match the script language, we could call these functions "InvalidIR" and "MalformedIR" (instead of "ValidIR" and "WellformedIR") and just negate the callsites?